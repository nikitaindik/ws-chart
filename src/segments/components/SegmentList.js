import React from 'react';
import Button from '../../components/Button';

import style from './SegmentList.module.css';

const SegmentList = ({ segments, activeSegmentId, onSegmentClick }) => (
  <div className={style.list}>
    {segments.map(segment => (
      <div key={segment.id}>
        <Button
          isActive={segment.id === activeSegmentId}
          onClick={() => onSegmentClick(segment.id)}
        >
          {segment.name}
        </Button>
      </div>
    ))}
  </div>
);

export default SegmentList;
