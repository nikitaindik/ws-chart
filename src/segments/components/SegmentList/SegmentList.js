import React from 'react';
import Button from '../../../components/Button';

const SegmentList = ({ segments, activeSegmentId, onSegmentClick }) => (
  <div>
    {segments.map(segment => (
      <div key={segment.id}>
        <Button
          isActive={segment.id === activeSegmentId}
          onClick={() => onSegmentClick(segment.id)}
        >
          {segment.name}
        </Button>
      </div>
    ))}
  </div>
);

export default SegmentList;
