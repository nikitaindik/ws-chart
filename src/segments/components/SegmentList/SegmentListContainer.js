import React, { useEffect } from 'react';
import { connect } from 'react-redux';

import { loadSegmentList, chooseSegment } from '../../actions';
import {
  selectSegments,
  selectIsListLoaded,
  selectActiveSegmentId,
} from '../../selectors';

import SegmentList from './SegmentList';

const SegmentListContainer = ({
  isListLoaded,
  loadList,
  chooseSegment,
  segments,
  activeSegmentId,
}) => {
  useEffect(() => {
    loadList();
  }, []);

  if (!isListLoaded) {
    return <div>Loading...</div>;
  }

  return (
    <SegmentList
      segments={segments}
      onSegmentClick={chooseSegment}
      activeSegmentId={activeSegmentId}
    />
  );
};

const mapStateToProps = state => ({
  segments: selectSegments(state),
  isListLoaded: selectIsListLoaded(state),
  activeSegmentId: selectActiveSegmentId(state),
});

const mapDispatchToProps = {
  loadList: loadSegmentList,
  chooseSegment,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SegmentListContainer);
