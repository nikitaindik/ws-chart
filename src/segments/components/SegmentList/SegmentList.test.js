import React from 'react';
import { render } from 'react-testing-library';

import SegmentList from './SegmentList';

test('SegmentList snapshot', () => {
  const mockedProps = {
    segments: [{ name: 'Ten', id: '10' }, { name: 'Twenty', id: '20' }],
    activeSegmentId: '10',
    onSegmentClick: jest.fn(),
  };

  const { container } = render(<SegmentList {...mockedProps} />);
  expect(container).toMatchSnapshot();
});
