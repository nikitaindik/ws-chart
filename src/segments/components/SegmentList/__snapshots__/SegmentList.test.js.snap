// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentList snapshot 1`] = `
<div>
  <div>
    <div>
      <button
        class="button active"
      >
        Ten
      </button>
    </div>
    <div>
      <button
        class="button"
      >
        Twenty
      </button>
    </div>
  </div>
</div>
`;
